smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java

```diff
@@ -0,0 +1,336 @@
+
+        // upload a new file
+        try {
+            File file = new File(FILE_NAME);
+            body = new MultipartBody.Builder().setType(MultipartBody.FORM)
+                    .addFormDataPart("upload", FILE_NAME,
+                            RequestBody.create(MediaType.parse("application/octet-stream"),
+                                    file))
+                    .build();
+
+            Response resp = postRequest(STACK.opennms().getBaseUrlExternal() + REST_FILESYSTEM + "/contents?f=" + FILE_NAME, "POST", body);
+            Assert.assertEquals(200, resp.code());
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        }
+
+        // update the file
+        body = new MultipartBody.Builder().setType(MultipartBody.FORM)
+                .addFormDataPart("upload", "<name>OpenNMS Smoke Test</name>")
+                .build();
+
+        try {
+            Response resp = postRequest(STACK.opennms().getBaseUrlExternal() + REST_FILESYSTEM + "/contents?f=" + FILE_NAME, "POST", body);
+            Assert.assertEquals(200, resp.code());
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        }
+
+        // get file
+        try {
+            Response resp = postRequest(STACK.opennms().getBaseUrlExternal() + REST_FILESYSTEM + "/contents?f=" + FILE_NAME, "GET", null);
+            Assert.assertEquals(200, resp.code());
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        }
+
+        // remove the file
+        MediaType mediaType = MediaType.parse("text/plain");
+        body = RequestBody.create(mediaType, "");
+        try {
+            Response resp = postRequest(STACK.opennms().getBaseUrlExternal() + REST_FILESYSTEM + "/contents?f=" + FILE_NAME, "DELETE", body);
+            Assert.assertEquals(200, resp.code());
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        }
+    }
+
+    /**
+     * Test that the File Editor will deny file that is not in the correct format
+     */
+    @Test
+    public void wrongFileName() {
+        LOG.info("Wrong file name test");
+
+        // upload a new file
+        try {
+            File file = new File("README.md");
+            RequestBody body = new MultipartBody.Builder().setType(MultipartBody.FORM)
+                    .addFormDataPart("upload", file.getName(),
+                            RequestBody.create(MediaType.parse("application/octet-stream"),
+                                    file))
+                    .build();
+
+            Response resp = postRequest(STACK.opennms().getBaseUrlExternal() + REST_FILESYSTEM + "/contents?f=" + FILE_NAME, "POST", body);
+            Assert.assertEquals(400, resp.code());
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        }
+    }
+
+    /**
+     * Test that the File Editor will deny changes that are not in the correct format
+     */
+    @Test
+    public void wrongUpdate() {
```

Comment:
May want a more specific method name / comment, for example `updateFailsOnXmlValidation`, "Test that the File Editor will deny changes when invalid XML is uploaded for an XML file"